crates/autopilot/src/decoded_settlement.rs

```diff
@@ -226,69 +236,74 @@ impl DecodedSettlement {
                     flags: trade.8.into(),
                     executed_amount: trade.9,
                     signature: trade.10,
-                })
-                .collect(),
+                };
+                match trade.compute_uid(domain_separator, &tokens) {
+                    Ok(order_uid) => {
+                        trade.order_uid = order_uid;
+                        Some(trade)
+                    }
+                    Err(err) => {
+                        tracing::error!(
+                            ?err,
+                            ?trade,
+                            "failed to calculate order uid, we don't know which order this trade \
+                             belongs to"
+                        );
+                        None
```

Comment:
In general I don't think settlement should be considered valid if this error happens (it basically means we have a bug somewhere in our code). See new implementation that is supposed to replace this whole file: https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/domain/settlement/mod.rs#L122-L123

However fine leaving it as is, considering that
1. old code worked in the same way
2. don't expect this to ever be hit
3. new code will be used instead, quite soon